File: frontend/src/models/auction.ts
export interface Venue {
    id: number;
    name: string;
    location?: string;
    description?: string;
    createdAt: string;
}

export interface Auction {
    id: number;
    venueId: number;
    auctionDate: string;
    startTime?: string;
    endTime?: string;
    status: 'scheduled' | 'in_progress' | 'completed' | 'cancelled';
    createdAt: string;
    updatedAt: string;
}

export interface AuctionItem {
    id: number;
    auctionId: number;
    fishermanId: number;
    fishType: string;
    quantity: number;
    unit: string;
    status: string;
    highestBid?: number;
    highestBidderId?: number;
    highestBidderName?: string;
    createdAt: string;
}

export interface Bid {
    itemId: number;
    buyerId: number;
    price: number;
}

File: backend/internal/server/handler/admin/response/auction.go
package response

import "time"

// Auction represents a detailed view of an auction for admins.
type Auction struct {
	ID          int       `json:"id"`
	VenueID     int       `json:"venue_id"`
	AuctionDate string    `json:"auction_date"`
	StartTime   *string   `json:"start_time"`
	EndTime     *string   `json:"end_time"`
	Status      string    `json:"status"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}
